pkg/policy/groups/helpers.go

```diff
@@ -65,9 +67,32 @@ func createDerivativeCNP(ctx context.Context, cnp *cilium_v2.CiliumNetworkPolicy
 		},
 	}
 
-	rules, err := cnp.Parse()
+	var (
+		rules api.Rules
+		err   error
+	)
+
+	if ccnpDerived {
+		// Temporary fix for CCNPs. See #12834.
+		// TL;DR. CCNPs are converted into SlimCNPs and end up here so we need to
+		// convert them back to CCNPs to allow proper parsing.
+		// For more details on implementation see - https://github.com/cilium/cilium/pull/12851
+		ccnp := &cilium_v2.CiliumClusterwideNetworkPolicy{
+			TypeMeta:            cnp.TypeMeta,
+			ObjectMeta:          cnp.ObjectMeta,
+			CiliumNetworkPolicy: cnp,
+			Status:              cnp.Status,
+		}
+
+		// Update the Kind of owner if the policy is CCNP derived.
+		derivativeCNP.ObjectMeta.OwnerReferences[0].Kind = cilium_v2.CCNPKindDefinition
+		rules, err = ccnp.Parse()
+	} else {
+		rules, err = cnp.Parse()
+	}
+
 	if err != nil {
-		return nil, fmt.Errorf("Cannot parse policies: %s", err)
+		return derivativeCNP, fmt.Errorf("Cannot parse policies: %s", err)
```

Comment:
Why was it necessary to change this?

Reply:
Nit:

```suggestion
		return derivativeCNP, fmt.Errorf("cannot parse policies: %v", err)
```

Reply:
https://github.com/cilium/cilium/blob/fb0c3e592d7a9da854f9d38345fbb5720ea3b073/pkg/policy/groups/actions.go#L149-L156 - Here in line 156 we use `derivativeCNP.ObjectMeta.Name`, now if `cnp.Parse()` fails and returns nil this will panic.